### ISSUE TITLE
VariableBuffer result polygon has incorrect hole

### ISSUE THREAD

**Issue by Alison-97** [locationtech/jts#998]
### Description

I am encountering an issue with the Variable Buffer class. Specifically, I am working with an open LineString. Given that the LineString is open and does not intersect itself, the buffer operation should yield a simple polygon without any interior rings (holes). However, in the current implementation, the resulting polygon contains an unnecessary interior ring. This is causing a disruption in the order of the resulting polygon coordinates.

### Steps to Reproduce
Here's the .Net code to reproduce the issue:
```
public void Test()
        {
            var coordinates = new Coordinate[]
            {
                new Coordinate(200         ,   91.37371939),
                new Coordinate(199.9906516 ,   71.396149),
                new Coordinate(199.925198  ,   66.39682954),
                new Coordinate(135.4977781 ,   0.125906357),
                new Coordinate(56.41194182 ,   0.598175835),
                new Coordinate(0.024520295 ,   69.50077743),
                new Coordinate(0.000508719 ,   74.50086084),
                new Coordinate(6.22E-22    ,   76.39546845),
            };
            double[] widths = new double[]
            {
                5.4573551091373282,
                5.5548897583660466,
                5.5793006151088935,
                6.0919286067086738,
                6.4781230068945561,
                6.96634014175149,
                6.9907509984943372,
                7,
            };

            var innerLine = new LineString(coordinates);

            var variableBuffer = new VariableBuffer(innerLine, widths);
            var completeBuffer = variableBuffer.GetResult();

        }
```

This is a plot of the inner line coordinates.
![image](https://github.com/locationtech/jts/assets/97237245/8e03f5d1-3052-48f2-91ea-9ec544f951db)

**Comment by dr-jts** [comment]
This is due to a flaw in the generation of the buffers of the individual segments.  In particular, the segment buffer for the last segment is being generated with a short spike in the buffer offset curve:
<img width="472" alt="image" src="https://github.com/locationtech/jts/assets/3529053/873f7a38-ed49-423f-aa15-9d34ea20e27d">

This overlaps the previous segment buffer polygon in a way that causes a small gap in the unioned buffers:
<img width="516" alt="image" src="https://github.com/locationtech/jts/assets/3529053/4ceb9f24-b6ed-41fd-82cd-84d56034e2cb">

Hopefully this is a straightforward fix to improve the buffer offset curve generation to prevent spikes.

**Comment by dr-jts** [comment]
The example geometry as WKT:
```
LINESTRING (200 91.37371939, 199.9906516 71.396149, 199.925198 66.39682954, 135.4977781 0.125906357, 56.41194182 0.598175835, 0.024520295 69.50077743, 0.000508719 74.50086084, 0 76.39546845)
```

**Comment by dr-jts** [comment]
@Alison-97 hopefully the fix in #1014 solves your issue.

Can I ask what you are using VariableBuffer for?
